### DISCUSSION TITLE
[QUESTION]: How tasks rollback for workflow failure?

### DISCUSSION THREAD

**Discussion by ladenjet** [Netflix/conductor#2979] [Q&A]
In my opinion, task definition fields should at least add the following two additional properties:
1. rollback policy, indicating whether rollback service should be executed when wokflow rollbacks for some reason.
3. rollback service, indicating the service name (usually in the form of RESTful request) to be called when rollback occurs.

Lets's say a group of tasks are defined as "taska-->taskb-->taskc",  when taskc fails, rollback logic should be executed in reverse sequence.  For this instance,  rollback service of each task should be called in the sequence like "taskc-->taskb-->taska".
 I cannot see such function by now, should I develop it myself?

**Comment by jxu-nflx** [comment]
Hello @ladenjet that is a very interesting proposal. However, conductor workflow and task don't support transactions/rollbacks on its own.

**Comment by zee-m8** [comment]
@jxu-nflx if conductor does not support transactions/rollbacks on its own, what in your opinion, would be the best way to implement it? I came across a very similar use-case as @ladenjet and wanted to know if this is even possible in conductor?
